org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -6389,59 +6389,19 @@ public static void setComplianceOptions(String compliance, Map options) {
 		long jdkLevel = CompilerOptions.versionToJdkLevel(compliance);
 		int major = (int) (jdkLevel >>> 16);
 		switch(major) {
-			case ClassFileConstants.MAJOR_VERSION_1_3:
```

Comment:
It's a pity this method doesn't declare any exceptions, since silently ignoring an unsupported compliance level could easily be problematic for clients that are caught unawares.

OTOH, it looks like we have been ignoring any kind of gibberish all along.